agent/requestor/src/market.rs

```diff
@@ -28,29 +32,31 @@ pub(crate) fn build_demand(
         },
     });
 
+    let common_constraint = Constraints::new_clause(
+        And,
+        vec![
+            ConstraintKey::new("golem.inf.mem.gib").greater_than(ConstraintValue::new(0.5)),
+            ConstraintKey::new("golem.inf.storage.gib").greater_than(ConstraintValue::new(1)),
+            ConstraintKey::new("golem.com.pricing.model").equal_to(ConstraintValue::new("linear")),
+        ],
+    );
     let subnet_constraint = match subnet {
         Some(subnet) => {
             log::info!("Using subnet: {}", subnet);
             properties.as_object_mut().unwrap().insert(
                 "golem.node.debug.subnet".to_string(),
                 serde_json::Value::String(subnet.clone()),
             );
-            format!("(golem.node.debug.subnet={})", subnet.clone())
+            common_constraint.and(Constraints::new_single(
+                ConstraintKey::new("golem.node.debug.subnet").equal_to(ConstraintValue::new(serde_json::Value::String(subnet.clone())))
+            )).to_string()
         }
-        None => "".to_string(),
+        None => common_constraint.to_string(),
     };
 
     Demand {
         properties,
-        constraints: format!(
-            "(&
-            (golem.inf.mem.gib>0.5)
-            (golem.inf.storage.gib>1)
-            (golem.com.pricing.model=linear)
-            {}
-        )",
-            subnet_constraint
-        ),
+        constraints: subnet_constraint,
```

Comment:
Just a note: I would do common_constraint.to_string() here, instead of in line 54. It would leave constraint object open for modification and adding new constraints.